pkg/images/defaults/flavor.go

```diff
@@ -272,20 +291,20 @@ func GetImageFlavorByName(flavorName string, isReleaseBuild bool) (ImageFlavor,
 // roxctl should instead rely on different ways to determine which image defaults to use. Such as asking users to
 // provide a command-line argument.
 func GetImageFlavorFromEnv() ImageFlavor {
-	envValue := strings.TrimSpace(imageFlavorEnv())
-	if envValue == "" && !buildinfo.ReleaseBuild {
-		envValue = ImageFlavorNameDevelopmentBuild
-		log.Warnf("Environment variable %s not set, this will cause a panic in release build. Assuming this code is executed in unit test session and using '%s' as default.", imageFlavorEnvName, ImageFlavorNameDevelopmentBuild)
-	}
+	envValue := GetImageFlavorNameFromEnv()
 	f, err := GetImageFlavorByName(envValue, buildinfo.ReleaseBuild)
 	if err != nil {
 		// Panic if environment variable's value is incorrect to loudly signal improper configuration of the effectively
 		// build-time constant.
-		log.Panicf("Incorrect image flavor in environment variable %s: %s", imageFlavorEnvName, err)
+		panicImageFlavorEnv(err)
 	}
 	return f
 }
 
+func panicImageFlavorEnv(err error) {
+	log.Panicf("Incorrect image flavor in environment variable %s: %s", imageFlavorEnvName, err)
```

Comment:
@msugakov Can this panic cause our components to crash loop when not expected?
If yes, should this be a utils.Should to panic in development but not on prod builds?